packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/document_migrator.ts

```diff
@@ -175,6 +141,21 @@ export class DocumentMigrator implements VersionedTransformer {
     return [document, ...additionalDocs];
   }
 
+  /**
+   * Returns true if the provided document has a higher version that the `targetTypeVersion`
+   * (defaulting to the last known version), false otherwise.
+   */
+  public isDowngradeRequired(
+    doc: SavedObjectUnsanitizedDoc,
+    { targetTypeVersion }: IsDowngradeRequiredOptions = {}
+  ): boolean {
+    if (!this.migrations) {
+      throw new Error('Migrations are not ready. Make sure prepareMigrations is called first.');
+    }
+    const typeMigrations = this.migrations[doc.type];
+    return downgradeRequired(doc, typeMigrations?.latestVersion ?? {}, targetTypeVersion);
+  }
```

Comment:
This is just exposing a wrapper around the `downgradeRequired` static utility, as the utility can't be used directly (given it requires access to the build migration map that only the doc migrator has)